### ISSUE TITLE
F3 lab is missing many old features

### ISSUE THREAD

**Issue by Goober5000** [scp-fs2open/fs2open.github.com#5994]
Many features have not yet been re-implemented after the lab was migrated to imgui.  The current lab is pretty much just a ship and mission background viewer.  The old lab allowed the tweaking of a whole bunch of settings, allowing many experiments to be performed without restarting the game engine.  (This was well-suited to its designation as a "lab".)

Here are some screenshots from 19.0.0.

Ships:

![lab_ship](https://github.com/scp-fs2open/fs2open.github.com/assets/1878523/45be0d8f-a841-49c5-876b-963048c74018)

Weapons:

![lab_weapons](https://github.com/scp-fs2open/fs2open.github.com/assets/1878523/a0aa9b7a-5f88-4b31-a214-937acf12694c)

**Comment by MjnMixael** [comment]
Having been all up in the Lab's business recently, I looked into this because this issue is the bug report version of not using grammar or paragraphs by just offering a couple overfilled screenshots and no helpful text as to what the actual work required would be.

I checked out and built 19.0.0 so I could look at the old lab code in debugger to see what exactly it allowed and how it worked. I'll go over each dialog window here one by one. Use the below info to decide if this issue is worth resolving or closing.

**tldr: This request seems mostly focused on re-creating the Object Flags and Object Variables dialogs but I think that serves very little utility while requiring a ton of work to do right. Personally I think the lab should be more for testing and not for tinkering as it is now.**

Dialogs in the old lab & their status in the new lab:
- Ship/Weapon class selection: ✅ 
- Background selection: ✅ 
- Render Options: ✅ _I did not do a 1:1 comparison of every option here but it seems mostly intact at a glance with some changes to match how models are rendered in PBR. Old lab was definitely designed with the old pre-PBR renderer in mind._
- Material Override: ❌ _Is this actually useful? I expect not. Personally I'd skip bothering with this at all. It's clutter and doesn't actually provide anything useful from a developer perspective except in maybe some exceedingly niche cases._
- Tech Description: ❌ _Probably not really necessary. Not sure it's more helpful here than in the Tech room. Would be trivial to do but I fail to see the point. Also, you can see the tech description in the currently implemented table viewer._
- Object Flags: ⚠️_Read is implemented via the table viewer. Write is not implemented. More details below**_
- Object Variables: ⚠️_Read is implemented via the table viewer. Write is not implemented. More details below**_

**As an active FSO mod developer, I question the actual utility of these two (Flags and Variables dialogs).**

Not only was the old behavior buggy (I could not reliably get the variables editor to work without setting a breakpoint to allow an extra frame to pass and thus actually get me into edit mode before the frame loop returned early and keypresses were ignored) but I was able to quickly and easily get the game to assert and crash by entering bad data. Now, the old lab being buggy is not reason enough alone to not do something similar with the new lab (though it does cause me to wonder who, if anyone, really used these features. I suspect the number is in the single digits).

Additionally, the benefit of setting class data during runtime bypasses all the checks we have in place during parsing to prevent bad data. Seems like a nice footgun and a possible headache inducing source of bug reports in the future unless some in-depth safety checks are added for each variable that is allowed to be changed. That feels like a lot of work and upkeep depending on just how many of the ship/weapon variables that would be supported here and how much SCP wants to commit to supporting future values as well. Ships and weapons are not getting less complex and this could end up as an absolutely huge list of variables and flags.

Furthermore, it would need to be made clear that changing these values will affect things outside the Lab. I could see a real scenario where players mess around with values unaware of the effects, go back to playing missions, and then end up with strange behavior that they would be reporting to mod authors. The lack of clarity would make such situations frustrating and confusing for all parties involved.

Finally and most importantly, this would need a very clear undo system because it changes the values for the object class globally and it will stay that way permanently until the game is restarted or the user changes the values manually back to the original using the Lab again. At the very least a system would need to be designed to keep an unaltered copy of any ship/weapon info object that is changed so that an object could be reset because to parsed values.

Putting all of that together and this feels like a lot of overhead when a simple table edit and game restart does the trick. I've done some really tedious coding in FSO for small gains (HUD Config 😅 ) and this is a bit much even for me.

Further details on what the unimplemented dialogs actually did:
- Material Override: _Allowed setting the various rendered textures (diffuse, shine, glow, etc) to solid RGB colors._
- Tech Description: _This was a simple dialog that showed the tech description and nothing else._
- Object Flags: _Showed a list of all available ship/weapon flags with a checkbox allowing to toggle them on or off._
- Object Variables: _Showed a semi-complete list of variables related to the object (things like speed, density, etc.. but also name, species index, etc) and if you knew the keypress to activate edit mode would allow changing these values._
